controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthContext.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.rpc.auth;
+
+import io.pravega.auth.AuthHandler;
+import lombok.Getter;
+
+import java.security.Principal;
+
+/**
+ * A gateway to objects held in gRPC {@link io.grpc.Context} by the {@link AuthInterceptor}, for the current gRPC
+ * request.
+ */
+public class AuthContext {
+
+    @Getter
+    private Principal principal;
+
+    @Getter
+    private AuthHandler authHandler;
+
+    private AuthContext(Principal principal, AuthHandler authHandler) {
+        this.principal = principal;
+        this.authHandler = authHandler;
+    }
+
+    /**
+     * Initiates a new instance using the current {@link io.grpc.Context}.
+     *u
+     * @return an instance containing {@link Principal} and {@link AuthHandler} from the current {@link io.grpc.Context}
+     */
+    public static AuthContext current() {
+        Principal principal = AuthInterceptor.AUTH_CONTEXT_TOKEN.get();
+        AuthInterceptor serverAuthInterceptor = AuthInterceptor.INTERCEPTOR_OBJECT.get();
```

Comment:
AuthInterceptor.AUTH_CONTEXT_TOKEN is static.
If there are multiple concurrent grpc requests, wouldnt we get the same auth context token?

Reply:
It represents only the key used to index the `Principal` object stored in the current gRPC context, which remains the same, and so it is static. The value - the `Principal` object - is a different object for each gRPC call, so concurrent requests don't share the values.

I've renamed it and the second key as well, and have now described them through comments as shown in the following fragment. Hopefully, the comments reveal the intent better.

https://github.com/pravega/pravega/blob/ec4427f74dc71e46414628d68208561d5ad9b76b/controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthInterceptor.java#L38-L48